TeachingRecordSystem/src/TeachingRecordSystem.Api/V3/Implementation/Operations/CreateTrnRequest.cs

```diff
@@ -33,181 +27,141 @@ public record CreateTrnRequestCommand
+        {
+            MatchedRecords = matchResult.Outcome switch
+            {
+                TrnRequestMatchResultOutcome.PotentialMatches =>
+                    matchResult.PotentialMatchesPersonIds
+                        .Select(id => new PostgresModels.TrnRequestMatchedRecord() { PersonId = id })
+                        .ToList(),
+                TrnRequestMatchResultOutcome.DefiniteMatch => [new PostgresModels.TrnRequestMatchedRecord() { PersonId = matchResult.PersonId }],
+                _ => []
+            }
+        };
 
-        trnToken = emailAddress is not null && trn is not null ? await trnRequestService.CreateTrnTokenAsync(trn, emailAddress) : null;
-        aytqLink = trnToken is not null ? trnRequestService.GetAccessYourTeachingQualificationsLink(trnToken) : null;
+        dbContext.TrnRequestMetadata.Add(trnRequestMetadata);
 
-        var contactId = Guid.NewGuid();
+        await dbContext.SaveChangesAsync();
+        txn.Complete();
 
-        var metadataMessage = CreateMetadataOutboxMessage(
-            potentialDuplicate,
-            resolvedPersonId: trn is not null ? contactId : null,
-            trnToken);
+        // This explicit Dispose() is to trigger the TransactionCompletedEvent *before* we call backgroundJobScheduler.WaitForJobToCompleteAsync()
+        txn.Dispose();
 
-        await crmQueryDispatcher.ExecuteQueryAsync(new CreateContactQuery()
+        if (jobId is not null)
         {
-            ContactId = contactId,
-            FirstName = firstName,
-            MiddleName = middleName,
-            LastName = command.LastName,
-            StatedFirstName = Option.Some(command.FirstName),
-            StatedMiddleName = Option.Some(command.MiddleName ?? ""),
-            StatedLastName = Option.Some(command.LastName),
-            DateOfBirth = command.DateOfBirth,
-            Gender = command.Gender?.ConvertToContact_GenderCode() ?? Contact_GenderCode.Notavailable,
-            EmailAddress = emailAddress,
-            NationalInsuranceNumber = NationalInsuranceNumber.Normalize(command.NationalInsuranceNumber),
-            ReviewTasks = reviewTasks,
-            ApplicationUserName = currentApplicationUserName,
-            Trn = trn,
-            TrnRequestId = TrnRequestService.GetCrmTrnRequestId(currentApplicationUserId, command.RequestId),
-            TrnRequestMetadataMessage = metadataMessage,
-            AllowPiiUpdates = allowContactPiiUpdatesFromUserIds.Contains(currentApplicationUserId.ToString())
-        });
+            using var backgroundJobCompletionCts = new CancellationTokenSource();
+            backgroundJobCompletionCts.CancelAfter(_waitForBackgroundJobCompletion);
+
+            await backgroundJobScheduler.WaitForJobToCompleteAsync(jobId, backgroundJobCompletionCts.Token)
+                .ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing | ConfigureAwaitOptions.ContinueOnCapturedContext);
+        }
```

Comment:
Is this the thing we discussed about waiting for the job to complete so the state is consistent? If the job is cancelled after timeout what are the consequences of that?